src/main/kotlin/lotto/domain/Lottery.kt

```diff
@@ -1,25 +1,22 @@
 package lotto.domain
 
 import lotto.domain.LottoMachine.Companion.LOTTO_PRICE
-import java.math.BigDecimal
 
 class Lottery(
-    lottos: List<Lotto>,
+    lottos: Lottos,
     winningLotto: Lotto,
     bonusNumber: LottoNumber,
 ) {
     private val prizes: List<Prize> =
-        lottos.map { lotto ->
-            Prize.calculate(winningLotto.compareMatches(lotto), lotto.contains(bonusNumber))
-        }
+        lottos
+            .compareAllTo(winningLotto)
+            .map { Prize.calculate(it.value, it.key.contains(bonusNumber)) }
 
     val result =
         Prize.entries.associateWith {
             prizes.count { prize -> it == prize }
         }
 
-    val returnRate: BigDecimal =
-        prizes.sumOf { it.value }
-            .toBigDecimal()
-            .divide(lottos.size.times(LOTTO_PRICE).toBigDecimal())
+    val returnRate: Double =
+        prizes.sumOf { it.value }.toDouble() / (lottos.size * LOTTO_PRICE)
```

Comment:
from: https://github.com/next-step/kotlin-lotto/pull/1143#discussion_r2025569299
> I know that intellij provides feature for type inlays but I just wanted to know if you have any specific circumstances where you specify types?

This depends on the situation!
I find IntelliJ's type Inlay hints to be quite useful, but since we can't use them on Github, I make an effort to help the code reviewers whenever I can. It's also helpful to specify types when using generics, especially for variances.

Reply:
I see. Thank you for the insights 😁